File: backend/tour-design-api/src/main/java/com/tourdesign/platform/service/TravelPackageService.java
package com.tourdesign.platform.service;

import com.tourdesign.platform.model.TravelPackageModel;
import java.util.List;
import java.util.Optional;

public interface TravelPackageService {

    List<TravelPackageModel> list();

    TravelPackageModel create(TravelPackageModel obj);

    Optional<TravelPackageModel> search(Long id);

    Optional<TravelPackageModel> searchDetail(Long id);

    Optional<TravelPackageModel> update(Long id, TravelPackageModel obj);

    boolean delete(Long id);
}

File: frontend/tour-design-web/src/app/services/travel-package.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { TravelPackageModel } from '../models/travel-package.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TravelPackageService {
  private baseUrl = '/travel-package';

  constructor(private api: ApiService) {}

  list(): Observable<TravelPackageModel[]> {
    return this.api.get<TravelPackageModel[]>(this.baseUrl);
  }

  search(id: number): Observable<TravelPackageModel> {
    return this.api.get<TravelPackageModel>(`${this.baseUrl}/${id}`);
  }

  create(obj: TravelPackageModel): Observable<TravelPackageModel> {
    return this.api.post<TravelPackageModel>(this.baseUrl, obj);
  }

  update(id: number, obj: TravelPackageModel): Observable<TravelPackageModel> {
    return this.api.put<TravelPackageModel>(`${this.baseUrl}/${id}`, obj);
  }

  delete(id: number): Observable<void> {
    return this.api.delete<void>(`${this.baseUrl}/${id}`);
  }
}
